### PR TITLE
[SYCL][Bindless] improve vulkan interop sampled images testing

### PR DESCRIPTION
**Pull request by cppchedy**
Improve the testing of Vulkan interop sampled images by extending the test to run with different combination of parameters such as image size, number of dimensions

### CODE REVIEW

**Review by Seanst98 — APPROVED** [review]

**Review by isaacault — APPROVED** [review]

**Review by przemektmalon — APPROVED** [review]

File: sycl/test-e2e/bindless_images/vulkan_interop/sampled_images.cpp [hunk 1]
```diff
@@ -231,15 +334,58 @@ bool run_test() {
...
+
+  valid &= run_test<3, int16_t, 1, sycl::image_channel_type::signed_int16,
+                    sycl::image_channel_order::r, class int16_3d>({64, 32, 64},
+                                                                  {4, 2, 4}, 0);
+
+  if (!valid) {
+    std::cout << "sampled images test has failure(s)\n";
+  } else {
+    std::cout << "sampled images test passes\n";
+  }
```
Comment (Seanst98) [inline sycl/test-e2e/bindless_images/vulkan_interop/sampled_images.cpp:386]: Why the switch from a simple `Test passed!` output to this? Or even `All tests passed!` as in our other consolidated testing?

If you decide to change this then can you also edit the `unsampled_images.cpp` file to print the same message.

**Comment by bjoernknafla** [comment]
Noticing the current [SYCL E2E on AWS CUDA test fail](https://github.com/intel/llvm/actions/runs/6554008514) for ` SYCL :: DeviceLib/built-ins/ext_native_math.cpp` also for other PRs.

File: sycl/test-e2e/bindless_images/vulkan_interop/sampled_images.cpp (hunk 1 above)
Comment (cppchedy) [reply sycl/test-e2e/bindless_images/vulkan_interop/sampled_images.cpp]: done.

**Comment by againull** [comment]
This fail on CUDA is not related and covered by https://github.com/intel/llvm/issues/11562
Failed Tests (1):
SYCL :: DeviceLib/built-ins/ext_native_math.cpp

**Pull request merged**